### PR TITLE
Cleanup Path tests

### PR DESCRIPTION
**Pull request by hughbe**
- Use xunit InlineData
- Follow some consistent test guidelines

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @JeremyKuhne

**Comment by JeremyKuhne** [comment]
@hughbe:
Thanks for cleaning up the inputs!

@stephentoub:
Did we state somewhere that we wanted all tests for a class to be in a single test class? I've been pushing the other way.

The number of tests we need for Path, Directory, FileInfo, etc. are significant- I'm a little concerned about navigating large test files and the impact to all of the tests running in serial (as xunit runs class members in serial).

**Comment by stephentoub** [comment]
> Did we state somewhere that we wanted all tests for a class to be in a single test class?

Nope.

**Comment by JeremyKuhne** [comment]
@hughbe 
Could you put the combine tests back in their own class? (For reasons stated above.)

I appreciate your effort here! Fyi: I'm also going through IO tests right now to try and fill out and unify some old tests for porting work I'm doing between Desktop and Core. As part of this I'm separating out tests for individual methods as we've done for a number of the System.IO tests.

**Comment by hughbe** [comment]
Yup, I'll do this. Sorry for changing it

**Comment by JeremyKuhne** [comment]
lgtm

**Comment by stephentoub** [comment]
Test Innerloop Ubuntu Debug Build and Test please
Test Innerloop OSX Debug Build and Test please

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please

**Comment by hughbe** [comment]
I don't think the test failure here is anything to do with my PR btw

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please

File: src/System.Runtime.Extensions/tests/System/IO/Path.Combine.cs [hunk 1]
```diff
@@ -2,288 +2,253 @@
...
-    {
-        //paths has 2 elements
-        Verify(new String[] { "abc", "def" });
-    }
-
-    [Fact]
-    public static void VerifyPathManyElements()
-    {
-        //paths has many elements
-        Verify(new String[] { "abc" + s_separator + "def", "def", "ghi", "jkl", "mno" });
```
Comment (stephentoub) [inline src/System.Runtime.Extensions/tests/System/IO/Path.Combine.cs]: What happened to this test case, i.e. an element that contains a separator and others that don't?

Comment (hughbe) [inline src/System.Runtime.Extensions/tests/System/IO/Path.Combine.cs]: Whoops. I added this back in, and rebased

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by hughbe** [comment]
@stephentoub, I had to rebase to fix c# 6 nits that I missed and undo the change adding a namespace that confused the PR

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Pull request merged**